src/frontend/src/views/Data/components/SamplesView/components/SampleTableModalManager/components/CreateNSTreeModal/index.tsx

```diff
@@ -100,18 +104,44 @@ export const CreateNSTreeModal = ({
   // Filter based on location
   const { data: groupInfo } = useGroupInfo();
   const { data: namedLocationsData } = useNamedLocations();
-  const namedLocations: NamedGisaidLocation[] =
-    namedLocationsData?.namedLocations ?? [];
+  const namedLocations: NamedGisaidLocation[] = useMemo(() => {
```

Comment:
I'm a little confused on this use of useMemo. Usually useMemo would be used when a function is computationally expensive. What is the reason for using it here?

Reply:
real reason: The browser ate it without the memo and the editor suggested it.
I'm not entirely sure why this kept getting recomputed.